cylc/flow/prerequisite.py

```diff
@@ -268,7 +273,10 @@ def satisfy_me(self, outputs: Iterable['Tokens']) -> 'Set[Tokens]':
             if prereq not in self._satisfied:
                 continue
             valid.add(output)
-            self[prereq] = 'satisfied naturally'
+            if self._satisfied[prereq] != 'satisfied naturally':
```

Comment:
This should be a straightforward merge with the Skip mode logic here, but we will need to make sure it's well tested to avoid mistakes.